src/ci_providers/provider_circleci.ts

```diff
@@ -30,22 +32,20 @@ function _getSHA(inputs: UploaderInputs): string {
 
 function _getSlug(inputs: UploaderInputs): string {
   const { args, environment: envs } = inputs
-  let slug = args.slug || ''
+
+  const slug = setSlug(
+    args.slug,
+    envs.CIRCLE_PROJECT_USERNAME,
+    envs.CIRCLE_PROJECT_REPONAME,
+  )
   if (slug !== '') {
     return slug
   }
-  if (envs.CIRCLE_PROJECT_REPONAME !== '') {
-    slug = `${envs.CIRCLE_PROJECT_USERNAME}/${envs.CIRCLE_PROJECT_REPONAME}`
-  } else {
-    if (envs.CIRCLE_REPOSITORY_URL) {
-      slug = `${envs.CIRCLE_REPOSITORY_URL.split(':')[1].split('.git')[0]}`
-    } else {
-      throw new Error(
-        'Unable to detect slug from env. Please set manually with the -r flag',
-      )
-    }
+
+  if (isSetAndNotEmpty(envs.CIRCLE_REPOSITORY_URL)) {
+    return `${envs.CIRCLE_REPOSITORY_URL?.split(':')[1].split('.git')[0]}`
```

Comment:
The reason the `?` is necessary here now is because the logic that once lived in the condition was moved to an outside function which currently conveys less type-level guarantees than the logic did by itself. Essentially, the truthiness of `envs.CIRCLE_REPOSITORY_URL` guaranteed that it was
- not of type `undefined`
- not of type `null`
- a `string` with a `length` property with a positive value

Together, those guarantees allowed for `split` to be called directly on `envs.CIRCLE_REPOSITORY_URL`.

Defining a type guard that correctly conveys the three logical conditions above is fairly difficult. It may be better to revert back to a simple truthiness check here:

```suggestion
  if (envs.CIRCLE_REPOSITORY_URL) {
    return `${envs.CIRCLE_REPOSITORY_URL.split(':')[1].split('.git')[0]}`
```